rts/motoko-rts/src/stream.rs

```diff
@@ -0,0 +1,24 @@
+//! The implementation of streaming serialisation
+//!
+//! When serialising Motoko stable variables to stable memory we used to first completely.
+//! fill up an in-heap buffer and then copy that wholesale into stable memory. This can be
+//! disadvategous for two reasons:
+//!  - double copying
+//!  - heap congestion (especially for the compactig collector)
+//!
+//! Instead now we'll 
+//!
+//!
+
+// Layout of a stream node:
```

Comment:
Looks like you are introducing a new object tag an type for this, but I wonder if we can get away with something much simpler. Shall we chat?

Reply:
Nope, I'll reuse the `blob` tag